ArduCopter/GCS_Mavlink.cpp

```diff
@@ -1062,6 +1062,38 @@ void GCS_MAVLINK_Copter::handleMessage(mavlink_message_t* msg)
             break;
 #endif
 
+#if AC_RALLY == ENABLED
+        case MAV_CMD_NAV_RALLY_POINT: //not including home
+
+            if (!is_zero(packet.param1)) { // Go to a specific rally ID
+                copter.rally.cmd_go_to_rally_id = true;
+                copter.rally.go_to_rally_id = (uint8_t)packet.param2;
+            } else if (!is_zero(packet.param3)) { // Go to nearest rally from loc except home
+                copter.rally.cmd_go_to_rally_only = true;
+            } else if (!is_zero(packet.param4)) { // Go to new rally location (and save it in EEPROM)
```

Comment:
Where are all these param definitions? I don't see them in MAVLink...

Reply:
There is nothing defined in MAVLink.. I can add a MAVLINK specific message to ardupilotmega.xml if you prefer but I don't see any sense to wait for MAVLINK to define all the messages before we can use them. When they'll be ready, we can always refractor

Reply:
Adding things to code before adding them to MAVLink is part of what has led to problems. Things should be discussed in MAVLink first then we can add them to code.